cwl/src/main/scala/cwl/CwlWomExpression.scala

```diff
@@ -85,12 +90,31 @@ final case class WorkflowStepInputExpression(input: WorkflowStepInput, override
     }
   }
 
-  override def evaluateFiles(inputTypes: Map[String, WomValue], ioFunctionSet: IoFunctionSet, coerceTo: WomType) = ???
+  override def evaluateFiles(inputTypes: Map[String, WomValue], ioFunctionSet: IoFunctionSet, coerceTo: WomType) =
+    "Programmer error: Shouldn't use WorkflowStepInputExpressions to find output files. You silly goose.".invalidNel
 
   override def inputs = graphInputs ++ input.source.toSet.flatMap{ inputSource: InputSource => inputSource match {
     case WorkflowStepInputSource.String(s) => Set(FullyQualifiedName(s).id)
     case WorkflowStepInputSource.StringArray(sa) => sa.map(FullyQualifiedName(_).id).toSet
   }}
 }
 
+final case class InitialWorkDirFileGeneratorExpression(entry: IwdrListingArrayEntry) extends CwlWomExpression {
+  override def cwlExpressionType: WomType = WomSingleFileType
+  override def sourceString: String = entry.toString
+
+  override def evaluateValue(inputValues: Map[String, WomValue], ioFunctionSet: IoFunctionSet): ErrorOr[WomValue] = entry match {
+    case IwdrListingArrayEntry.StringDirent(content, StringOrExpression.String(entryname), _) =>
+      Try(Await.result(ioFunctionSet.writeFile(entryname, content), Duration.Inf)).toErrorOr
+    case _ => ??? // TODO WOM and the rest....
+  }
+
 
+  override def evaluateFiles(inputTypes: Map[String, WomValue], ioFunctionSet: IoFunctionSet, coerceTo: WomType): ErrorOr[Set[WomFile]] =
+    "Programmer error: Shouldn't use InitialWorkDirRequirement listing to find output files. You silly goose.".invalidNel
```

Comment:
ToL: should this not conceptually return the file that this expression aims to create ?

Reply:
`evaluateFiles` is "examine this output expression and tell me what files I'm going to need to have delocalized from JES in order to evaluate it". 

So IMO it's not applicable to the InitialWorkDirRequirement expressions that create `File`s

Reply:
Ok I've always thought we were also using it to figure out the files that were going to be created prior to running the task (like by `write_lines` for example). Anyway, works for me 👍

Reply:
It's a side effect of PAPI needing to know delocalizations before we start running the job. If/when PAPI2 happens, this entire function will hopefully go away

Reply:
ToL I think we should encode `Input` vs `Output` expressions as types, perhaps in an output extends input manner.  Then we'd only evaluate the ones that have outputs.  I'll add an issue w/ these thoughts.